Examples/Io/Root/include/ActsExamples/Io/Root/RootEstimatedParametersWriter.hpp

```diff
@@ -0,0 +1,110 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "ActsExamples/EventData/Track.hpp"
+#include "ActsExamples/Framework/WriterT.hpp"
+
+#include <mutex>
+#include <vector>
+
+class TFile;
+class TTree;
+
+namespace ActsExamples {
+
+/// @class RootEstimatedParametersWriter
+///
+/// Write out the estimated track parameters from found seeds into a TTree
+///
+/// Safe to use from multiple writer threads - uses a std::mutex lock.
+///
+/// Each entry in the TTree corresponds to one set of estimated track parameters
+/// for optimum writing speed. The event number is part of the written data.
+///
+/// A common file can be provided for to the writer to attach his TTree,
+/// this is done by setting the Config::rootFile pointer to an existing
+/// file
+///
+/// Safe to use from multiple writer threads - uses a std::mutex lock.
+class RootEstimatedParametersWriter final
+    : public WriterT<TrackParametersContainer> {
+ public:
+  struct Config {
+    /// Input estimated track parameters collection.
+    std::string inputTrackParameters;
```

Comment:
It's indeed a bit nasty. I'd like to write out both the estimated track parameters and the truth particle info so that we could get a feeling of what is the resolution of the estimated track parameters.  Is there a better way for this purpose?

Reply:
Writing out reconstructed parameters and best match truth is perfectly fine, but I am wondering why you did not extend the existing `RootTrackParameterWriter`?

Reply:
OK, I added the changes on top of the `RootTrackParameterWriter`.